src/coreclr/jit/vartype.h

```diff
@@ -185,7 +185,7 @@ inline bool varTypeIsArithmetic(T vt)
 template <class T>
 inline bool varTypeIsGC(T vt)
 {
-    return ((varTypeClassification[TypeGet(vt)] & (VTF_GCR | VTF_BYR)) != 0);
+    return (TypeGet(vt) == TYP_REF) || (TypeGet(vt) == TYP_BYREF);
```

Comment:
Was this an intentional change? Looks like it's a cheaper and better change, but also seems a bit unrelated to the changes above.

Reply:
the intent of this PR was to slightly improve TP and this change seems to be doing so, we replace a memory load with two constants - it gives 0.1% improvement on all targets (although, a small regression on arm64 due to more instructions used for compare) as we eliminate the load